odoo/tools/mail.py

```diff
@@ -595,3 +595,30 @@ def formataddr(pair, charset='utf-8'):
                 name=email_addr_escapes_re.sub(r'\\\g<0>', name),
                 addr=address)
     return address
+
+
+def email_domain_extract(email):
+    """Return the domain of the given email."""
+    if not email:
+        return
+
+    email_split = getaddresses([email])
+    if not email_split or not email_split[0]:
+        return
+
+    *_, domain = email_split[0][1].rpartition('@')
```

Comment:
```suggestion
    _, domain = email_split[0][1].rsplit('@', 1)
```
?

Reply:
I'd rather not, I prefer the clarity of "pattern matching"-ish constructs (though it's not quite that).